### PR TITLE
SDP-466 data/httphandler: Update invite and OTP messages on the database

### PR DESCRIPTION
**Pull request by CaioTeixeira95**
### What

This PR adds the `sms_registration_message_template` and `otp_message_template` attributes to the `PATCH /organization` request. Also, was added the `otp_message_template` column to the `organizations` table. Removed the `organization_sms_registration_message_template_contains_tags_check` constraint to not enforce templating for the user.

### Why

Ability to have dynamic messages for the `Receiver Wallet Registration Invite` and `OTP message`.

### Known limitations

We need to use the message from the database when sending the invitation message. This will be added in the next PR.

#### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [x] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/serve/httphandler/receiver_send_otp_handler_test.go [hunk 1]
```diff
@@ -26,9 +27,11 @@ import (
 
 type mockMessengerClient struct {
 	mock.Mock
+	message message.Message
 }
 
 func (m *mockMessengerClient) SendMessage(message message.Message) error {
+	m.message = message
```
Comment (marcelosalloum) [inline internal/serve/httphandler/receiver_send_otp_handler_test.go:34]: 💡  I know this approach works, but this is an anti-pattern because it doesn't take advantage of the mock library very well. Since we use testify's mocks, we should rely on the testing methods for the assertions.

Something like this could help you get a copy of that object for your assertions:
```go
		var receivedMessage message.Message
		mockMessenger.On("SendMessage", mock.AnythingOfType("message.Message")).
			Return(nil).
			Once().
			Run(func(args mock.Arguments) {
				receivedMessage = args.Get(0).(message.Message)
			})
```

File: internal/db/migrations/2023-09-18.0-alter-organizations-table-invite-and-otp-messages.sql
Comment (marcelosalloum) [inline internal/db/migrations/2023-09-18.0-alter-organizations-table-invite-and-otp-messages.sql:1]: Don't forget to update the migration file name with the latest date.

**Review by marcelosalloum — APPROVED** [review]
LGTM! Plz, feel free to merge it after improving the mock approach, as I described below.

File: internal/serve/httphandler/receiver_send_otp_handler_test.go (hunk 1 above)
Comment (CaioTeixeira95) [reply internal/serve/httphandler/receiver_send_otp_handler_test.go:34]: Wooooowww this is awesome!! Thank you so much.

File: internal/serve/httphandler/profile_handler_test.go [hunk 2]
```diff
@@ -454,6 +459,162 @@ func Test_ProfileHandler_PatchOrganizationProfile(t *testing.T) {
 		assert.Equal(t, "-03:00", org.TimezoneUTCOffset)
 		assert.Equal(t, imgBuf.Bytes(), org.Logo)
 	})
+
+	t.Run("updates organization's SMS Registration Message Template", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (CaioTeixeira95) [reply internal/serve/httphandler/profile_handler_test.go:463]: Sorry. I didn't understand. Could you elaborate?

[the remark this answers is not in the document]
Comment (marcelosalloum) [reply internal/serve/httphandler/profile_handler_test.go:463]: I mean that when I read a chunk of code like the following, it's hard to follow because there are so many blank lines. A lot of these calls that are separated by blank lines are basically part of the same thing, so they could be organized more closely together.

Does that make sense?

https://github.com/stellar/stellar-disbursement-platform-backend/blob/a45313d2d69d5a18dcbfd7366ff87a706fdedf9c/internal/serve/httphandler/profile_handler_test.go#L477-L534

[the remark this answers is not in the document]
Comment (CaioTeixeira95) [reply internal/serve/httphandler/profile_handler_test.go:463]: Sure. I'll group related things. Thank you.
